src/svg/qsvggenerator.cpp

```diff
@@ -934,6 +952,17 @@ bool QSvgPaintEngine::end()
     Q_D(QSvgPaintEngine);
 
     d->stream->setString(&d->defs);
+
+    for (auto it = clip_path_to_id.begin(); it != clip_path_to_id.end(); it++)
+    {
+        std::string path = it->first;
+        int path_id = it->second;
+
+        *d->stream << "<clipPath id=\"clip" << path_id << "\">" << '\n';
+        *d->stream << "\t" << path.c_str() << " \"/> \n";
+        *d->stream << "</clipPath>" << '\n';
+    }
```

Comment:
I would still prefer it if this was moved to stream to <defs> as the clip path is created. Everything else in this method is just closing tags, doesn't make sense to have an exception for clip paths.

You can still also keep the `clip_path_to_id` mapping for your lookup to avoid duplicates, which seems like a useful thing to do.

Reply:
Done

Reply:
Yeah this is looking nicer 👍